File: Happy_Number.java
/**
 * Problem: https://leetcode.com/problems/happy-number/
 */

public class Solution {
    public boolean isHappy(int n) {
        return isHappy(n, new HashSet<Integer>());
    }
    
    private boolean isHappy(int n, HashSet<Integer> set) {
        if(set.contains(n)) return false;
        if(n == 1) return true;
        set.add(n);
        int sum = 0;
        while(n > 0) {
            sum += (n%10) * (n%10);
            n /= 10;
        }
        return isHappy(sum, set);
    }
}

File: Swift/happy-number.swift
/**
 * https://leetcode.com/problems/happy-number/
 * 
 * 
 */ 
class Solution {
    func isHappy(_ n: Int) -> Bool {
        var n = n
        var list: Set<Int> = [n]
        while n > 1 {
            var sum = 0
            while n > 0 {
                sum += (n % 10) * (n % 10)
                n /= 10
            }
            if list.contains(sum) { return false }
            list.insert(sum)
            n = sum
        }
        return true
    }
}
